storage/watchable_store.go

```diff
@@ -240,7 +236,7 @@ func (s *watchableStore) syncWatchers() {
 		limit := cap(w.ch) - len(w.ch)
 		// the channel is full, try it in the next round
 		if limit == 0 {
-			nws = append(nws, w)
+			nws[w] = struct{}{}
```

Comment:
The current code has [`nws = append(nws, w)`](https://github.com/coreos/etcd/blob/master/storage/watchable_store.go#L229). Why would we delete here??